cms/djangoapps/contentstore/management/commands/swap_from_auto_track_cohort_pilot.py

```diff
@@ -0,0 +1,192 @@
+
+            # If there are no errors iterate over and update all of the items that had the access changed
+            if not errors:
+                for item in items_to_update:
+                    module_store.update_item(item, ModuleStoreEnum.UserID.mgmt_command)
+                    module_store.publish(item.location, ModuleStoreEnum.UserID.mgmt_command)
+
+            # Check if we should delete any partition groups if there are no errors.
+            # If there are errors, none of the xblock items will have been updated,
+            # so this section will throw errors for each partition in use
+            if partitions_to_delete and not errors:
+                partition_service = PartitionService(rerun_course_key)
+                course = partition_service.get_course()
+                for partition_to_delete in partitions_to_delete:
+                    # Get the user partition, and the index of that partition in the course
+                    partition = partition_service.get_user_partition(partition_to_delete.partition_id)
+                    partition_index = course.user_partitions.index(partition)
+                    group_id = int(partition_to_delete.group_id)
+
+                    # Sanity check to verify that all of the groups being deleted are empty,
+                    # since they should have been converted to use enrollment tracks instead.
+                    # Taken from contentstore/views/course.py.remove_content_or_experiment_group
+                    usages = GroupConfiguration.get_partitions_usage_info(module_store, course)
+                    used = group_id in usages
+                    if used:
+                        errors.append("Content group '%s' is in use and cannot be deleted."
+                                      % partition_to_delete.group_id)
+
+                    # If there are not errors, proceed to update the course and user_partitions
+                    if not errors:
+                        # Remove the groups that match the group ID of the partition to be deleted
+                        # Else if there are no match groups left, remove the user partition
+                        matching_groups = [group for group in partition.groups if group.id == group_id]
+                        if matching_groups:
+                            group_index = partition.groups.index(matching_groups[0])
+                            partition.groups.pop(group_index)
+                            # Update the course user partition with the updated groups
+                            course.user_partitions[partition_index] = partition
+                        else:
+                            course.user_partitions.pop(partition_index)
+                        module_store.update_item(course, ModuleStoreEnum.UserID.mgmt_command)
+
+            # If there are any errors, join them together and raise the CommandError
+            if errors:
+                raise CommandError("\n".join(errors))
+
```

Comment:
I do like the idea of printing out some confirmation messages (including intermediary messages). Especially since we are potentially operating on multiple courses at once; if one of them raises a CommandError, it would be good to be clear about which others did complete.